bundles/org.openhab.binding.siemensrds/src/main/java/org/openhab/binding/siemensrds/internal/RdsHandlerFactory.java

```diff
@@ -103,10 +106,13 @@ private synchronized void createDiscoveryService(RdsCloudHandler handler) {
      */
     private synchronized void destroyDiscoveryService(RdsCloudHandler handler) {
         // fetch the respective thing's service registration from our list
-        ServiceRegistration<?> serviceReg = discos.remove(handler.getThing().getUID());
+        @Nullable
+        Object object = discos.remove(handler.getThing().getUID());
+
+        // retrieve the respective discovery service
+        if (object instanceof ServiceRegistration<?>) {
```

Comment:
Ok, instead of using `instanceof` to do your null check, I suggest instead to just change the map type to
`private final Map<ThingUID, @Nullable ServiceRegistration<?>> discos = new HashMap<>();`

Reply:
Ok.